homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,368 @@
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.info("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    def get_supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            features = 0
+            try:
+                # Map features for better reading
+                if self._bulbtype.features.brightness:
+                    features = features | SUPPORT_BRIGHTNESS
+                if self._bulbtype.features.color:
+                    features = features | SUPPORT_COLOR
+                if self._bulbtype.features.effect:
+                    features = features | SUPPORT_EFFECT
+                if self._bulbtype.features.color_tmp:
+                    features = features | SUPPORT_COLOR_TEMP
+                return features
+            except WizLightNotKnownBulb:
+                _LOGGER.info(
+                    "Bulb is not present in the library. Fallback to full feature"
+                )
+                return SUPPORT_FEATURES_RGB
+        # fallback
+        return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+
+    @property
+    def device_info(self):
+        """Get device specific attributes."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "name": self._attr_name,
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+            if self._light.state is None:
+                self.update_state_unavailable()
+            else:
+                self.update_state_available()
```

Comment:
Move this in an `else` block of the `try` since it cannot fail.

Reply:
I was not aware of the try -> else feature 😉